test/integration/submit_pledge_test.rb

```diff
@@ -6,6 +6,9 @@ class SubmitPledgeTest < ActionDispatch::IntegrationTest
     @user = create :user
     @patient = create :patient
```

Comment:
I thought you could set attributes after creation. I'll take another look tonight but glad you got it to pass.